[DOCUMENTATION][feature/otx] Explanation section. PR 2 (#1566)
open-edge-platform/training_extensions: docs/source/explanation/algorithms/index.rst

.. _algo_section_ref:

OTX supports different training types to solve a variety of computer vision problems. This section provides what exactly we utilize inside our algorithms providing an end-to-end solution to solve real-life computer vision problems.

To this end we support:

> - **Supervised training**. Supervised learning for computer vision involves training a model on a labeled dataset of images. The model learns to associate specific features in the images with the corresponding labels. This is the most common approach for computer vision tasks such as object detection and image classification.

Reviewer: ```suggestion
- **Supervised training**. This is the most common approach for computer vision tasks such as object detection and image classification. Supervised learning involves training a model on a labeled dataset of images. The model learns to associate specific features in the images with the corresponding labels. 
```

> - **Incremental learning**. OTX provides functionality to let the model to learn from new data as it becomes available and improve its performance over time, rather than having to retrain the entire model every time new data is added. OTX supports also the class incremental approach. In this approach, the model is first trained on a set of classes, and then incrementally updated with new classes of data, while keeping the previously learned classes' knowledge. The class incremental approach is particularly useful in situations where the number of classes is not fixed and new classes may be added over time.

Reviewer: ```suggestion
- **Incremental learning**. This learning approach lets the model train on new data as it becomes available, rather than retraining the entire model on the whole dataset every time new data is added. OTX supports also the class incremental approach. In this approach, the model is first trained on a set of classes, and then incrementally updated with new classes of data, while keeping the previously learned classes' knowledge. The class incremental approach is particularly useful in situations where the number of classes is not fixed and new classes may be added over time.
```
